src/runner.sml

```diff
@@ -1,24 +1,35 @@
-(* TODO *)
-type RunResult = unit
+structure RunnerOption = struct
+  type RunnerOption = { sequenced: bool, seed: int option }
 
-signature RUNNER =
-sig
-  datatype RunnerOption =
-      Seed of int
+  datatype RunnerOption
+    = Seed of int option
     | Sequenced
 
-  val runWithOptions: RunnerOption list -> Test.Test -> RunResult
-  val run: Test.Test -> RunResult
+  val empty = { seed = NONE, sequenced = false }
+
+  fun build opts =
+    List.foldl (fn (opt, acc) =>
+      case opt of
+        Seed seed =>
+          { seed = seed
+          , sequenced = (#sequenced acc) }
+      | Sequenced =>
+          { seed = (#seed acc)
+          , sequenced = true }
+    ) empty opts
 end
 
-structure Runner: RUNNER =
+structure Runner  =
 struct
+  infix  3 |>     fun x |> f = f x
```

Comment:
what is this

Reply:
This operator is there because we don't have the pipe operator by default.